### PR TITLE
Add information on extension footprint analyzer

### PR DESCRIPTION
**Pull request by ishakhsuvarov**
## Purpose of this pull request

This pull request adds information about Extension Footprint analyzer feature of Magento Marketplace. 

## Affected DevDocs pages

-  https://devdocs.magento.com/marketplace/sellers/extension-footprint.html

whatsnew
Added a new topic describing [Extension Footprint Analysis](https://devdocs.magento.com/marketplace/sellers/extension-footprint.html) on the Magento Marketplace.

### CODE REVIEW

File: src/marketplace/sellers/code-sniffer.md [hunk 1]
```diff
@@ -7,6 +7,8 @@ title: Code Sniffer
 
 Code Sniffer is a static test that uses static code analysis to detect violations of the [Magento Coding Standard](https://github.com/magento/magento-coding-standard/) to prevent common coding errors.
 
+Limited information, like number of errors, warnings and coding standards version used are displayed in the Quality Report section of product lising page on the Magento Marketplace.
```
Comment (BarnyShergold) [inline src/marketplace/sellers/code-sniffer.md:10]:
```suggestion
Limited information, like the number of errors, warnings and the coding standards version used are displayed in the Quality Report section of product listing page on the Magento Marketplace.
```

File: src/marketplace/sellers/extension-footprint.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
+---
+group: marketplace-sellers
+title: Extension Footprint analysys
```
Comment (BarnyShergold) [inline src/marketplace/sellers/extension-footprint.md:3]:
```suggestion
title: Extension Footprint Analysis
```

File: src/marketplace/sellers/extension-footprint.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+---
+
+{: .bs-callout .bs-callout-info}Footprint analyzer is currently released as a beta feature meaning that some of the information may not be copletely accurate or present at all. Beta status of the feature is also indicated on the product lisging page.
```
Comment (BarnyShergold) [inline src/marketplace/sellers/extension-footprint.md:6]: Put text on next line. It should not be on the same line as content in braces.

Comment (BarnyShergold) [inline src/marketplace/sellers/extension-footprint.md:6]:
```suggestion
{: .bs-callout .bs-callout-info}The footprint analyzer is currently released as a beta feature, meaning that some of the information may not be completely accurate or present at all. Beta status of the feature is also indicated on the product listing page.
```

File: src/marketplace/sellers/extension-footprint.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+## Overview
+
+Foorptint analysis is an automated check that generates report based on static analysis of the extension code. After processing, reports are visible on the product listing page and include information on certain specific features which provide Magento Marketplace customers insights on extension capabilities from the code perspective.
```
Comment (BarnyShergold) [inline src/marketplace/sellers/extension-footprint.md:10]:
```suggestion
Footprint analysis is an automated check that generates a report based on the static analysis of the extension code. After processing, reports are visible on the product listing page and include information on certain specific features which provide Magento Marketplace customers insights on extension capabilities from the code perspective.
```

File: src/marketplace/sellers/extension-footprint.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+## When testing is done
+
+All extension packages have their footprint analyzed at the time of submissions. This check is not a test and therefore cannot be passed or failed.
```
Comment (BarnyShergold) [inline src/marketplace/sellers/extension-footprint.md:14]:
```suggestion
All extension packages have their footprint analyzed at the time of submission. This check is not a test and therefore cannot be passed or failed.
```

File: src/marketplace/sellers/extension-footprint.md [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+## What is being checked
+
+At the moment Footptint Analyzer check is performing static package code testing to identify APIs exposed by the module. Report usually consists of the following sections:
```
Comment (BarnyShergold) [inline src/marketplace/sellers/extension-footprint.md:18]:
```suggestion
At the moment the footprint analyzer check is performed on the static package code to identify APIs exposed by the module. The report usually consists of the following sections:
```

File: src/marketplace/sellers/extension-footprint.md [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+*  Programmatic API and Data interfaces – provides information on the number of programmatic APIs found in the extension code package.
+*  Web API – provides insights on the number of REST and SOAP APIs exposed by the extension package.
+*  GraphQL – verifies if extension exposes any GraphQL entities.
```
Comment (BarnyShergold) [inline src/marketplace/sellers/extension-footprint.md:22]:
```suggestion
*  GraphQL – verifies if the extension exposes any GraphQL entities.
```

**Review by BarnyShergold — CHANGES_REQUESTED** [review]
Nice new addition

File: src/marketplace/sellers/code-sniffer.md [hunk 8]
```diff
@@ -7,6 +7,8 @@ title: Code Sniffer
 
 Code Sniffer is a static test that uses static code analysis to detect violations of the [Magento Coding Standard](https://github.com/magento/magento-coding-standard/) to prevent common coding errors.
 
+Limited information, like the number of errors, warnings and the coding standards version used are displayed in the Quality Report section of product listing page on the Magento Marketplace.
```
Comment (jeff-matthews) [inline src/marketplace/sellers/code-sniffer.md:10]:
```suggestion
Limited information is displayed in the Quality Report section of the product listing page on the Magento Marketplace, including the number of errors, warnings, and the coding standards version that was used.
```

File: src/marketplace/sellers/extension-footprint.md [hunk 9]
```diff
@@ -0,0 +1,22 @@
+---
+group: marketplace-sellers
+title: Extension Footprint Analysis
+---
+
+{: .bs-callout .bs-callout-info} The footprint analyzer is currently released as a beta feature, meaning that some of the information may not be completely accurate or present at all. Beta status of the feature is also indicated on the product listing page.
```
Comment (jeff-matthews) [inline src/marketplace/sellers/extension-footprint.md:6]:
```suggestion
{:.bs-callout-info}
The footprint analyzer is a beta feature, meaning that some of the information may not be completely accurate or present at all. The beta status of the feature is also indicated on the product listing page.
```

File: src/marketplace/sellers/extension-footprint.md [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+
+## Overview
+
+Footprint analysis is an automated check that generates a report based on the static analysis of the extension code. After processing, reports are visible on the product listing page and include information on certain specific features which provide Magento Marketplace customers insights on extension capabilities from the code perspective.
```
Comment (jeff-matthews) [inline src/marketplace/sellers/extension-footprint.md:10]:
```suggestion
Footprint analysis is an automated check that generates a report based on the static analysis of the extension code. After processing, reports are visible on the product listing page and include information on certain specific features, which provide Magento Marketplace customers insights on extension capabilities from the code perspective.
```

File: src/marketplace/sellers/extension-footprint.md [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+
+## When testing is done
+
+All extension packages have their footprint analyzed at the time of submission. This check is not a test and therefore cannot be passed or failed.
+
+## What is being checked
+
+At the moment the footprint analyzer check is performed on the static package code to identify APIs exposed by the module. The report usually consists of the following sections:
```
Comment (jeff-matthews) [inline src/marketplace/sellers/extension-footprint.md:18]:
```suggestion
At the moment, the footprint analyzer check is performed on the static package code to identify APIs exposed by the module. The report usually consists of the following sections:
```

File: src/marketplace/sellers/extension-footprint.md [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+
+*  Programmatic API and Data interfaces – provides information on the number of programmatic APIs found in the extension code package.
```
Comment (jeff-matthews) [inline src/marketplace/sellers/extension-footprint.md:20]:
```suggestion
*  Programmatic API and data interfaces—Provides information on the number of programmatic APIs found in the extension code package.
```

File: src/marketplace/sellers/extension-footprint.md [hunk 13]
```diff
@@ -0,0 +1,22 @@
...
+*  Web API – provides insights on the number of REST and SOAP APIs exposed by the extension package.
```
Comment (jeff-matthews) [inline src/marketplace/sellers/extension-footprint.md:21]:
```suggestion
*  Web API—Provides insights on the number of REST and SOAP APIs exposed by the extension package.
```

File: src/marketplace/sellers/extension-footprint.md [hunk 14]
```diff
@@ -0,0 +1,22 @@
...
+*  GraphQL – verifies if the extension exposes any GraphQL entities.
```
Comment (jeff-matthews) [inline src/marketplace/sellers/extension-footprint.md:22]:
```suggestion
*  GraphQL—Verifies if the extension exposes any GraphQL entities.
```

**Review by jeff-matthews — CHANGES_REQUESTED** [review]
Thanks for the new topic @ishakhsuvarov!

I've requested additional changes to get this ready for publication. 

Please also add a new entry for this new topic in the table of contents (TOC) file. It probably makes sense to add it near the existing "Code Sniffer" entry: https://github.com/magento/devdocs/blob/master/src/_data/toc/marketplace-sellers.yml#L110 

Or maybe it makes more sense to add it further down near the "checks" since this footprint analyzer isn't technically a test. What do you think?

**Comment by ishakhsuvarov** [comment]
Thank you for your review and fixes @jeff-matthews 

I've added it TOC as the next entry after semantic version check. I think it's a good place for it since it's still a part of technical review even though it's not a test in traditional meaning.

**Review by jeff-matthews — APPROVED** [review]
Looks good. Thanks @ishakhsuvarov!

**Comment by jeff-matthews** [comment]
@magento import code to magento-devdocs/devdocs

**Comment by magento-engcom-team** [comment]
@jeff-matthews the branch with code successfully imported into`magento-devdocs/devdocs` repository. Branch name: `imported-magento-devdocs-8962`.

**Pull request merged**